### ISSUE TITLE
Allow extracting statistics

### ISSUE THREAD

**Issue by tech234a** [PatNeedham/google-it#37]
I would like to be able to programmatically access search stats via google-it (basically retrieving and interpreting the "Page 4 of about 37,700,000 results (0.67 seconds)" text), especially the result count and search time. The div id is "resultStats". Thanks!

**Comment by PatNeedham** [comment]
Thank you for expressing interest in this! I've actually thought about it myself as well. Especially that total results count text because that could help enable a true pagination capability. But also the results which aren't necessarily the "search results" but rather the tangential info which appears like the featured snippet, "People also ask" list, "Videos" list, "Images" list, "Top stores" list, and Related searches links. [Tesla engine](https://www.google.com/search?q=tesla+engine) is an example query whose results page has all of those.

This is a pretty good milestone for v2 of the google-it library, since I want the default behavior for it to return all of this stuff by default, instead of requiring an extra config argument. But I'll start with the "Page X of Y results (Z seconds)" text since that's almost always going to appear on the page. @tech234a what do you think would be a good way to represent that info in a JSON object? I'm preferring something like 
```
{
  results: [...], // original list that's currently being returned
  stats: {
    page: 4,
    approximateResults: 37700000,
    seconds: 0.67
  }
}
```

**Comment by tech234a** [comment]
That would be helpful, thanks!

**Comment by PatNeedham** [comment]
This is now a work in progress, and will be part of v1.5.0. [This commit](https://github.com/PatNeedham/google-it/commit/a7b1d72e5bdac9ea149c4d2b6dd4bdabcb4da78d)  captures the additions made to enable it. The existing optional `diagnostics` field can be set to true inside the options object passed to the `googleIt` function. And just like the HTML selectors for result titles, links, and snippets, this selector (defined in the utils file) can be overridden via environment variable (`GOOGLE_IT_RESULT_STATS_SELECTOR`) or by optional field passed to the options object as well (`resultStatsSelector`). Same goes for the cursor selector, used to access the current page:
![image](https://user-images.githubusercontent.com/967811/73229219-fc9ec080-4146-11ea-9b48-f0c556bcfc95.png)

I'm going to try to come up with two extra unit tests to cover the cases of:
1) that result stats part of the screen doesn't exist
2) the cursor at the bottom doesn't exist

Before opening the PR. But feel free to try it out from that branch if you want. You can install an npm module [directly from a Github branch](https://stackoverflow.com/a/39732501/772985). For example,
```
npm install git://github.com/patneedham/google-it.git#feature/allow-extracting-statistics --save
```

**Comment by PatNeedham** [comment]
It's merged to master branch now, with the additional test to cover a result page without those sections present. This is what the stats object looks like in that scenario: `{ page: NaN, approximateResults: '', seconds: NaN }`. I'll eventually need to include that on a documentation/wiki page somewhere so it's clear, but it's here for now.

**Comment by PatNeedham** [comment]
Now available in v1.5.0
